module/metrics/transaction.go

```diff
@@ -81,11 +84,67 @@ func NewTransactionCollector(transactionTimings mempool.TransactionTimings, log
 			Subsystem: subsystemTransactionSubmission,
 			Help:      "counter for the success/failure of transaction submissions",
 		}, []string{"result"}),
+		scriptExecutedDuration: promauto.NewHistogramVec(prometheus.HistogramOpts{
+			Name:      "script_executed_duration",
+			Namespace: namespaceAccess,
+			Subsystem: subsystemTransactionSubmission,
+			Help:      "histogram for the duration in ms of the round trip time for executing a script",
+			Buckets:   []float64{1, 100, 500, 1000, 2000, 5000},
+		}, []string{"script_size"}),
+		transactionResultDuration: promauto.NewHistogramVec(prometheus.HistogramOpts{
+			Name:      "transaction_result_fetched_duration",
+			Namespace: namespaceAccess,
+			Subsystem: subsystemTransactionSubmission,
+			Help:      "histogram for the duration in ms of the round trip time for getting a transaction result",
+			Buckets:   []float64{1, 100, 500, 1000, 2000, 5000},
+		}, []string{"script_size"}),
+		scriptSize: promauto.NewHistogramVec(prometheus.HistogramOpts{
+			Name:      "script_size",
```

Comment:
I'm not sure. I watched one video that recommended using labels for this sort of thing. I think they said something like otherwise you end up with too many metrics. You could end up with 10 metrics instead of 1 metric with 10 labels. We would only have two right now though.

With multiple metrics do you know if it's easy to combine them into one histogram? I feel like being able to see the distribution of script size across all methods at the same time would be useful.

Reply:
I agree with what you said but had the following concerns:
1. A transaction is not really a script. So may be we can the label "payload_size" ?
2. Also, if reducing the number of metrics is better then can't the first two metrics - transaction result duration and script execution duration be combine? (I don't mind it either ways).

Reply:
Ok I think using two metrics makes sense. I've pushed those changes.